source/Halibut.Tests/Builders/ServiceFactoryBuilder.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using Halibut.ServiceModel;
+using Halibut.Tests.Util;
+
+namespace Halibut.Tests.Builders
+{
+    public class ServiceFactoryBuilder
+    {
+        bool _conventionVerificationDisabled;
+        DelegateServiceFactory factoryWithConventionVerification = new();
+        NaiveDelegateServiceFactory factoryWithNoConventionVerification = new();
+        
```

Comment:
Why register everything in one factory, when you can register in two!

I started writing something to store the factory Func<T> and the Contract and ClientContract types, then register them in the `Build` method, but it was way easier to just keep a copy of each, and try to register in all of them - `Build` just returns the one you want!

So a pretty dumb implementation, but it's simple and we can improve it later.